libraries/lib-graphics/graphics/Point.h

```diff
@@ -0,0 +1,192 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  Point.h
+
+  Dmitry Vedenko
+
+**********************************************************************/
+#pragma once
+
+#include <cmath>
+#include <type_traits>
+#include <numeric>
+
+namespace graphics
+{
+
+//! A point in 2D space.
+template <typename DataType>
+struct PointType final
```

Comment:
I would make a type distinction between points and vectors.  Then type checking will not allow meaningless operations.

Two vectors can be added or subtracted to make a vector.  A vector can be scaled by a number.

Addition of a point to a point or scaling of a point does not make sense.  Subtracting a point from a point does make sense, but the result is not a point, but a vector.

Reply:
So essentially, the points are meaningless; vectors can be used all over the place :-)